81.md

```diff
@@ -0,0 +1,50 @@
+NIP-81
+======
+
+Event Copy
+----------
+
+`draft` `optional` `author:arthurfranca`
+
+An event of `kind 1006` is used to copy any event from another author (except other copies). A copy
+allows its creator to keep content around even when the original gets deleted.
+
+The `.content` of a copy event SHOULD be the stringified JSON of the whole original event,
+optionally with extra fields inside a `.meta` key.
+The `.content` MAY be empty to represent a deleted event.
+
+The copy event MUST include a `published_at` tag with the same value from the original event
+or with the original event's `created_at` value if its `published_at` tag is absent.
+
+The copy event SHOULD include the following tags so to make them queriable:
+
+- Capital `I` tag with original event's `id`;
+- Capital `P` tag with original event's `pubkey`;
+- Capital `K` tag with original event's `kind`;
+- All original event's single-letter tags.
+
+`I`, `P` and `K` tags are reserved. They cannot be used by any other event kind.
```

Comment:
Example: if I'm copying an event that already has one P tag that also is a pubkey value, although with a custom meaning specific to that event kind, i end up with:
```
{
  // other fields
  "kind": 1006,
  "tags": [
    // other tags,
    ["P", "pubkeyA"], // copied from original event's "pubkey" field
    ["P", "pubkeyX"]  // copied from original event's one-letter tags
  ]
}
```

Now if I were to request one of the copies I made, and wanted a copy that had originally the `pubkey` field value "pubkeyX", I do `["REQ", "xyz", "{ \"kind\": 1006, \"#P\": [ \"pubkeyX\"], \"K\": .... }]`. This will wrongly bring me the above event.

That's why ideally we consider I, P and K in events other than copies invalid. Don't copy these invalid events and you will not have to filter things out client-side.

For the same reason I didn't make it a parameterized replaceable event (the original could alread have a "d" tag and its worse cause maybe only the first d tag is indexed on some relay implementations, cause of what is written on NIP-33)

This is similar to the problem [NIP-10](https://github.com/nostr-protocol/nips/blob/master/10.md) introduced. Can't request direct descendants of a specific root event without filtering out a bunch of events I don't want just because although the `e` tag means event id, it is used with different contexts ("root", "reply" and the legacy "mention").